sklearn/cluster/tests/test_hierarchical.py

```diff
@@ -751,3 +751,17 @@ def test_dist_threshold_invalid_parameters():
         AgglomerativeClustering(n_clusters=None,
                                 distance_threshold=1,
                                 compute_full_tree=False).fit(X)
+
+
+def test_invalid_shape_precomputed_dist_matrix():
+    rng = np.random.RandomState(0)
+    # Test non square distance matrix
+    X = rng.rand(5, 3)
+    with pytest.raises(
+        ValueError,
+        match="Matrix should be square, "
+    ):
+        AgglomerativeClustering(
+            affinity='precomputed',
+            linkage='complete'
+        ).fit(X)
```

Comment:
Maybe not here but it can probably be made a bit more compact

Reply:
More compact, done the same as previous tests in the file.